src/content/learn/understanding-your-ui-as-a-tree.md

```diff
@@ -0,0 +1,268 @@
+
+## The Render Tree {/*the-render-tree*/}
+
+A major feature of components is the ability to compose components of other components. As we [nest components](/learn/your-first-component#nesting-and-organizing-components), we have the concept of parent and child components, where each parent component may itself be a child of another component. 
+
+When we render a React app, we can model this relationship in a tree, known as the render tree.
+
+Here is a React app that renders a random greeting. 
+
+<Sandpack>
+
+```js App.js
+import Title from './Title';
+import Greeting from './Greeting';
+import Copyright from './Copyright';
+
+export default function App() {
+  return (
+    <>
+      <Title text="Generate a random greeting" />
+      <Greeting>
+        <Copyright year={2004} />
+      </Greeting>
+    </>
+  );
+}
+
+```
+
+```js Title.js
+export default function Title({fancy, text}) {
+    return <h1 className={fancy ? 'fancy' : ''}>{text}</h1>;
+}
+```
+
+```js Greeting.js
+import * as React from 'react';
+import greetings from './greetings';
+import Title from './Title';
+
+function randIndex(max) {
+  return Math.floor(Math.random() * max);
+}
+
+export default function Greeting({children}) {
+  const [index, setIndex] = React.useState(randIndex(greetings.length));
+  const greeting = greetings[index];
+  const nextGreeting = () => setIndex(randIndex(greetings.length));
+  return (
+    <>
+      <Title fancy text={greeting} />
+      <button onClick={nextGreeting}>🎲</button>
+      {children}
+    </>
+  );
+}
+```
+
+```js Copyright.js
+export default function Copyright({year}) {
+  return <p>©️ {year}</p>;
+}
+```
+
+```js greetings.js
+export default ['Howdy!', 'Hello2!', 'Hello3!', 'Hello3!', 'Hello4!'];
+```
+
+```css
+.fancy {
+    font-style: italic;
+    color: blue;
+}
+```
+
+</Sandpack>
+
+<Diagram name="render_tree" height={300} width={509} alt="Tree graph with 5 nodes. The root of the tree is App, with two arrows extending from it to nodes Greeting and Title. The arrows are labelled renders. Greeting node also has two arrows extended pointing down to nodes Title and Copyright.">
+
+React creates a UI tree made of components rendered, known as a render tree.
+
+</Diagram>
+
+From the example app, we can construct the above render tree. Each node in the tree represents a component and the root node is the [root component](/learn/importing-and-exporting-components#the-root-component-file). In this case, the root component is `App` and it is the first component React renders. Each arrow in the tree points from a parent component to a child component.
+
+We often refer to the components near the root of the tree as "top-level components". They are ancestors and have a lot of descendent components. Components that have no children are referred to as "leaves". 
```

Comment:
Is this important to what you're trying to communicate here? I think this concept might be better introduced closer to where someone can see the term in use (e.g. you should move your state to the lowest level, or "leaves" of your components.) rather than a blanket statement that is likely to be forgotten.

Reply:
My intention in bringing it up here is to mention it in the purpose of render trees -- or why they're useful.

I have this later on:

> Although render trees may differ across render pases, these trees are generally helpful for identifying what the top-level components are in a React app. Top-level components affect the rendering performance of all the components beneath them and often contain the most complexity.

I could move this introduction of 'top-level components' and 'leaves' closer to this.